packages/flutter_tools/lib/src/commands/install.dart

```diff
@@ -50,6 +51,11 @@ class InstallCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts
     if (device == null) {
       throwToolExit('No target device found');
     }
+
+    if (const LocalPlatform().operatingSystem.toLowerCase() == device?.name.toLowerCase()) {
```

Comment:
This logic looks very suspect to detect a desktop device.  Can the device itself say whether it supports installation?

https://github.com/flutter/flutter/blob/997d43618dc73b2d01a86adaccf9dc9672953c03/packages/flutter_tools/lib/src/device.dart#L566

Reply:
https://github.com/flutter/flutter/pull/118421/commits/d12e9a651a02ad077a1981d44efb13845577a915 should have those revisions.